addons/web/static/src/webclient/user_menu/user_menu.xml

```diff
@@ -23,7 +23,7 @@
                         <CheckBox
                             t-if="element.type == 'switch'"
                             value="element.isChecked"
-                            className="'form-switch d-flex flex-row-reverse justify-content-between p-0'"
+                            className="'form-switch d-flex flex-row-reverse justify-content-between w-auto p-0'"
```

Comment:
This way we're **always** overriding the SCSS introduced by https://github.com/odoo/odoo/commit/3cc9836400b89e09fe869b9d990ca678b0391f59 while keeping it.
If `width: fit-content;` is confirmed to be always "wrong", then it should be removed 😉

@chgo-odoo , have you check if `w-auto` produces the same visual result of `fit-content` for the issues that https://github.com/odoo/odoo/pull/127118 aimed to solve? 
@juliusc2066 , is there a specific reason for using `fit-content`?  

Thanks :)

Reply:
Hi @stefanorigano 👋

We are overriding the changes from https://github.com/odoo/odoo/commit/3cc9836400b89e09fe869b9d990ca678b0391f59 only in that navbar so this is enough regarding the issue from the pad IMO.

I did check if `w-auto` could be applied to every checkbox but it can't because it would not produce the expected result.


TL;DR -> `fit-content` should only be applied to checkboxes that are not inside `<button>` element IMO.

### Some interesting notes : 

> It could be nice to make a distinction between checkboxes inside a button or not inside a button.

For example, it makes sense for the dark mode switch to be able to toggle the switch no matter where you click on the button.

On the other side, in Community > Invoicing > Payment terms, we do have checkboxes and removing `width: fit-content` doesn't break anything, in fact, the checkbox can only be activated if your cursor is on it -> expected behaviour.


### **My guess is that this PR seems to only affect `form-switch` (?)**
For example in Invoicing, a `.o-checkbox.form-switch` could be activated from the whole input `width` before the changes applied in https://github.com/odoo/odoo/commit/3cc9836400b89e09fe869b9d990ca678b0391f59 

<img width="539" alt="image" src="https://github.com/odoo/odoo/assets/128030743/725bd941-2bbe-4d99-9b06-bd2956d27ac3">

This is related to the element above, `.o_field_widget` (in our case) that has a `width: 100%` After these changes, it can no more be activated from far away so that works as expected.

Could be nice to have the explanation from @juliusc2066 to better understand how that fix was created.